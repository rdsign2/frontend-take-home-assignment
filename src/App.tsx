import * as React from 'react';
import SavingGoalPlanSimulator from './pages/SavingGoalPlanSimulator';
import Wrapper from './components/Wrapper';
import './assets/styles/app.scss';

const App: React.FunctionComponent = () => {
  React.useEffect(() => {
    const themeMode = JSON.parse(window.localStorage.getItem('darkMode'));

    const changeTheme = (theme: string) => {
      document.getElementById('origin').setAttribute('data-theme', theme);
    };

    if (themeMode) {
      changeTheme('dark');
    } else {
      changeTheme('light');
    }

    // https://egghead.io/lessons/react-store-values-in-localstorage-with-the-react-useeffect-hook
    // Esse vídeo explica direitinho como fazer. Vou precisar controlar o tema pela STORE, para que o useeffect escute as alterações
    // Ver useselector e usedispatch
  }, []);

  return (
    <div id="origin" className="origin" data-theme="light">
      <Wrapper>
        <SavingGoalPlanSimulator />
      </Wrapper>
    </div>
  );
};

export default App;
