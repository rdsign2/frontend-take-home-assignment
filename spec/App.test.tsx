import * as React from 'react';
import { shallow } from 'enzyme';
import App from '../src/App';

describe('App', () => {
  const component = shallow(<App />);

  it('renders', () => {
    expect(component.exists()).toEqual(true);
  });

  it('html structure must match snapshot', () => {
    expect(component.getElements()).toMatchSnapshot();
  });
});
