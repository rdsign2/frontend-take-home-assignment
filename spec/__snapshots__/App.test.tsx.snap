// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App html structure must match snapshot 1`] = `
Array [
  <div
    className="origin"
    data-theme="light"
    id="origin"
  >
    <Wrapper>
      <SavingGoalPlanSimulator />
    </Wrapper>
  </div>,
]
`;
